nixos/modules/programs/direnv.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, config, pkgs, ... }: {
+
+  options.programs.direnv.enable = lib.mkOption {
+    type = lib.types.bool;
+    default = false;
+    example = true;
+    description = ''
+      Whether to enable direnv integration. Takes care of both installation and
+      setting up the sourcing of the shell. Additionally enables nix-direnv
+      integration. Note that you need to logout and login for this change to apply.
+    '';
+  };
+
+  config = lib.mkIf config.programs.direnv.enable {
+
+    environment.systemPackages = with pkgs; [
+      direnv
+      nix-direnv
+    ];
+
+    environment.pathsToLink = [
+      "/share/direnv"
+    ];
+
+    environment.sessionVariables.DIRENV_CONFIG = "/run/current-system/sw/share/direnv";
```

Comment:
As mentioned: This disallows user customization entirely, I think. 
Either we need a change to direnv to address this or we need a change here to give us the ability for the user to specify additional contents.